arkouda/dataframe.py

```diff
@@ -977,6 +980,140 @@ def rename(self, mapper: Union[Callable, dict], inplace: bool = False) -> Union[
             return obj
         return None
 
+    @typechecked
+    def _rename_index(
+        self, mapper: Union[Callable, Dict], inplace: bool = False
+    ) -> Optional[DataFrame]:
+        """
+        Rename indexes within the dataframe
+
+        Parameters
+        ----------
+        mapper : callable or dict-like
+            Function or dictionary mapping existing indexes to new indexes.
+            Nonexistent names will not raise an error.
+        inplace: bool
+            Default False. When True, perform the operation on the calling object.
+            When False, return a new object.
+        Returns
+        -------
+            DateFrame when `inplace=False`
+            None when `inplace=True`
+        See Also
+        -------
+            ak.DataFrame._rename_column
+            ak.DataFrame.rename
+        Notes
+        -----
+            This does not function exactly like pandas. The replacement value here must be
+            the same type as the existing value.
+        """
+        obj = self if inplace else self.copy()
+        if callable(mapper):
+            for i in range(obj.index.size):
+                oldval = obj.index[i]
+                newval = mapper(oldval)
+                if type(oldval) != type(newval):
+                    raise TypeError("Replacement value must have the same type as the original value")
+                obj.index.values[obj.index.values == oldval] = newval
+        elif isinstance(mapper, dict):
+            for key, val in mapper.items():
+                if type(key) != type(val):
+                    raise TypeError("Replacement value must have the same type as the original value")
+                obj.index.values[obj.index.values == key] = val
+        else:
+            raise TypeError("Argument must be callable or dict-like")
+        if not inplace:
+            return obj
+        return None
+
+    @typechecked
+    def rename(
+        self,
+        mapper: Optional[Union[Callable, Dict]] = None,
+        index: Optional[Union[Callable, Dict]] = None,
+        column: Optional[Union[Callable, Dict]] = None,
+        axis: Union[str, int] = 0,
+        inplace: bool = False,
+    ) -> Optional[DataFrame]:
+        """
+        Rename indexes or columns according to a mapping.
+
+        Parameters
```

Comment:
Just a nit but you might want to put the parameters in the docstring in the same sequence as in the function definition (i.e. column <-> index).